src/Spryker/Zed/AppPayment/Business/Payment/Cancel/CancelPayment.php

```diff
@@ -62,8 +62,10 @@ public function cancelPayment(CancelPaymentRequestTransfer $cancelPaymentRequest
         }
 
         /** @phpstan-var \Generated\Shared\Transfer\CancelPaymentResponseTransfer */
-        return $this->getTransactionHandler()->handleTransaction(function () use ($cancelPaymentRequestTransfer, $cancelPaymentResponseTransfer): \Generated\Shared\Transfer\CancelPaymentResponseTransfer {
-            $this->savePayment($cancelPaymentRequestTransfer->getPaymentOrFail(), $cancelPaymentResponseTransfer->getStatusOrFail());
+        return $this->getTransactionHandler()->handleTransaction(function () use ($cancelPaymentRequestTransfer, $cancelPaymentResponseTransfer): CancelPaymentResponseTransfer {
+            $paymentTransfer = $cancelPaymentResponseTransfer->getPayment() ?? $cancelPaymentRequestTransfer->getPaymentOrFail();
```

Comment:
it's the same as it was before if you change use `$cancelPaymentRequestTransfer->getPaymentOrFail()`

Reply:
Not quite getting your comment.

When the implementation sets the Payment on the `CancelPaymentResponseTransfer` then this PaymentTransfer will be used if not the original set one will be used.